src/app/terminal.js

```diff
@@ -284,39 +307,46 @@ class Terminal {
             // Received RTS
             case Broadcast.types.RTS: {
                 if (receivedBroadcast.destination === this && !this.sentData) {
+                    this.rtsTimeAccumulator=0;
+                    this.nextRtsTime+=receivedBroadcast.data[1];
                     this.queueBroadcast(Broadcast.types.CTS, receivedBroadcast.source, [receivedBroadcast.id,receivedBroadcast.data]);
                 }
                 break;
             }
 
             // Received CTS
             case Broadcast.types.CTS: {
+                this.rtsTimeAccumulator=0;
+                this.nextRtsTime+=receivedBroadcast.data[1];
                 if (receivedBroadcast.destination === this) {
                     const index = this.unackedRtses.findIndex(id => receivedBroadcast.data[0] === id);
                     if (index !== -1) {
                         this.unackedRtses.splice(index, 1);
+                        this.dataTimeout=receivedBroadcast.data[1]/numberOfAttempts;
+                        this.broadcastQueue=[];
```

Comment:
You're modifying what this.broadcastQueue points to instead of emptying the array in-place. I'm not sure how well this will work with it being passed as default parameter.

Reply:
isn't it the same? can you check?

Reply:
You have to check how it works with https://github.com/soryy708/networks-lab/blob/master/src/app/terminal.js#L88 and specifically with the expression `queue = this.broadcastQueue`
Does it contain the new queue or the old one?